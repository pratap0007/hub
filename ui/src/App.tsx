import React from 'react';
import './App.css';
import Category from './components/Category'
import {observer} from 'mobx-react';

const App = observer(({store}: any) => (
  <div className="App">
    <Category store={store} />
  </div>
))


export default App;