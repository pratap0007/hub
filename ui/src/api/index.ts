import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

interface Token {
  token: string;
  refreshInterval: string;
  expiresAt: number;
}

interface TokenData {
  access: Token;
  refresh: Token;
}

export interface AuthResponse {
  data: TokenData;
}

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
  resourceVersion(resourceId: number): Promise<IVersion>;
  versionUpdate(versionId: number): Promise<IVersion>;
  authentication(authCode: string): Promise<AuthResponse>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async authentication(authCode: string) {
    try {
      return axios.post(`${API_URL}/auth/login?code=${authCode}`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resourceVersion(resourceId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async versionUpdate(versionId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/version/${versionId}`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }
}
