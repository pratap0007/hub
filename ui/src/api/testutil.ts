import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
