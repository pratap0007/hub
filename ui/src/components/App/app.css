#root {
  height: 100vh;
}

.pf-c-check__label {
  color: white;
}
