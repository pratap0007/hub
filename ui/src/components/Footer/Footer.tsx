import React from 'react';
import { Card, Grid, Text, GridItem, TextVariants, TextContent } from '@patternfly/react-core';
import '@patternfly/react-core/dist/styles/base.css';
import tekton from '../../assets/logo/tekton.png';
import './Footer.css';

const Footer: React.FC = () => {
  return (
    <React.Fragment>
      <Card className="footer-card">
        <Grid>
          <GridItem span={12} className="text">
            <a href="https://cd.foundation">
              <img src="https://tekton.dev/partner-logos/cdf.png" alt="tekton.dev" />
            </a>
          </GridItem>
          <GridItem span={12} className="text">
            <TextContent className="text-color">
              <Text component={TextVariants.h3}>
                Tekton is a{' '}
                <Text component={TextVariants.a} href="https://cd.foundation">
                  Continuous Delivery Foundation
                </Text>{' '}
                project.
              </Text>
            </TextContent>
          </GridItem>
          <GridItem span={12} className="logo-margin">
            <img src={tekton} alt="Tekton" className="logo-size" />
          </GridItem>
          <GridItem span={2}></GridItem>
          <GridItem span={8} className="text-color">
            <Text>
              © 2020 The Linux Foundation®. All rights reserved. The Linux Foundation has registered
              trademarks and uses trademarks. For a list of trademarks of The Linux Foundation,
              please see our{' '}
              <Text
                component={TextVariants.a}
                href="https://www.linuxfoundation.org/trademark-usage/"
              >
                Trademark Usage page
              </Text>
              . Linux is a registered trademark of Linus Torvalds.{' '}
              <Text component={TextVariants.a} href="https://www.linuxfoundation.org/privacy/">
                Privacy Policy
              </Text>{' '}
              and{' '}
              <Text component={TextVariants.a} href="https://www.linuxfoundation.org/terms/">
                Terms of Use
              </Text>{' '}
              .
            </Text>
          </GridItem>
          <GridItem span={2}></GridItem>
        </Grid>
      </Card>
    </React.Fragment>
  );
};

export default Footer;
