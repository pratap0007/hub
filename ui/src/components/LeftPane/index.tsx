import React from 'react';
import { useObserver } from 'mobx-react';
import { GridItem, Grid } from '@patternfly/react-core';
import CatalogFilter from '../../containers/CatalogFilter';
import KindFilter from '../../containers/KindFilter';
import CategoryFilter from '../../containers/CategoryFilter';
import Sort from '../../containers/SortDropDown';
import './LeftPane.css';

const LeftPane: React.FC = () => {
  return useObserver(() => (
    <Grid hasGutter className="hub-leftpane">
      <GridItem span={8}>
        <Sort />
      </GridItem>
      <GridItem>
        <KindFilter />
      </GridItem>
      <GridItem>
        <CatalogFilter />
      </GridItem>
      <GridItem>
        <CategoryFilter />
      </GridItem>
    </Grid>
  ));
};
export default LeftPane;
