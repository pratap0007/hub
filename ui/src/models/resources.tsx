import { types, flow, getEnv, Instance, getSnapshot } from 'mobx-state-tree';


const Catalog = types.model("Catalog", {
    id: types.optional(types.number, 0),
    type: types.optional(types.string, ""),
})

export const Tag = types.model("Tag", {
    id: types.optional(types.number, 0),
    name: types.optional(types.string, " ")


})

export const LatestVersion = types.model("Version", {
    id: types.number,
    version: types.optional(types.string, "0.1"),
    displayName: types.string,
    description: types.string,
    minPipelinesVersion: types.optional(types.string, ""),
    rawURL: types.string,
    webURL: types.string,
    updatedAt: types.string,
})

export const ResourceItem = types.model("ResourceItem", {
    id: types.number,
    type: types.string,
    name: types.string,
    catalog: Catalog,
    latestVersion: LatestVersion,
    tags: types.optional(types.array(Tag), []),
    rating: types.optional(types.number, 0),

})

export const Resources = types.model("Resources", {
    resourceList: types.optional(types.array(ResourceItem), []),
    isLoading: types.optional(types.boolean, false)
})
    .views((self) => ({
        get count() {
            return self.resourceList.length;
        },
        get fetch() {
            return getEnv(self).fetch || window.fetch.bind(self);
        },
    }))
    .actions((self) => ({
        add(item: IResourceItem) {
            self.resourceList.push(item)

        }

    }))
    .actions((self) => ({
        load: flow(function* () {
            console.log("7777", { self })

            try {
                const resp = yield window.fetch('/resources.json').then((res: any) => res.json())
                resp.forEach((item: IResourceItem) => { self.add(item) })
                console.log(getSnapshot(self.resourceList))
            } catch (error) {
                console.error("Failed to fetch resources", error)
            }

        })
    }))
    .actions((self) => ({
        afterCreate() {
            self.load();
        },

    }))

export type IResourceItem = Instance<typeof ResourceItem>