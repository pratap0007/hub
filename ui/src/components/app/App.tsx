import React, { useState } from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import {
  Page,
  PageSection,
  Grid,
  GridItem,
  Pagination,
  Card,
  CardHeader,
  GalleryItem,
  CardBody,
  Gallery
} from '@patternfly/react-core';
import Footer from '../footer/Footer';
import Header from '../header/Header';
import Background from '../background/Background';
import './app.css';

const App: React.FC = () => {
  const [pageNumber, setPageNumber] = useState(1);
  const [perPgae, setPerPage] = useState(20);

  const setPage = (event: React.MouseEvent | React.KeyboardEvent | MouseEvent, page: number) => {
    setPageNumber(page);
  };
  const perPageSelect = (
    event: React.MouseEvent | React.KeyboardEvent | MouseEvent,
    perpage: number
  ) => {
    setPerPage(perpage);
  };

  return (
    <React.Fragment>
      <Page header={<Header />}>
        <Background />
        <PageSection>
          <Grid hasGutter>
            <GridItem span={1} rowSpan={1}>
              Filter
            </GridItem>
            <GridItem span={11} rowSpan={2}>
              {/* TODO: pagination should be  should be in Resource container componnet */}
              <Pagination
                itemCount={200}
                perPage={perPgae}
                onSetPage={setPage}
                onPerPageSelect={perPageSelect}
                page={pageNumber}
                isCompact
              />
              <Gallery hasGutter>
                {Array.apply(0, Array(9)).map((x, i) => (
                  <GalleryItem key={i}>
                    <Card style={{ backgroundColor: 'white' }}>
                      <CardBody>This is a card</CardBody>
                    </Card>
                  </GalleryItem>
                ))}
              </Gallery>
            </GridItem>
          </Grid>
        </PageSection>
        <Footer />
      </Page>
    </React.Fragment>
  );
};

export default App;
