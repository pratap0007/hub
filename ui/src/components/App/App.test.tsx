import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import { FakeHub } from '../../api/testutil';
import { CategoryStore } from '../../store/category';
import { KindStore } from '../../store/kind';
import { CatalogStore } from '../../store/catalog';
import { ResourceStore } from '../../store/resources';
import LeftPane from '../LeftPane/LeftPane';
import Footer from '../Footer/Footer';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);

describe('App', () => {
  it('should find the LeftPane component and match the count', () => {
    const store = ResourceStore.create(
      {},
      {
        api,
        kindStore: KindStore.create({}),
        catalogStore: CatalogStore.create({}),
        categoryStore: CategoryStore.create({}, { api })
      }
    );
    const component = shallow(<App store={store} />);

    expect(component.find(LeftPane).length).toEqual(0);
  });

  it('should find the Footer component and match the count', () => {
    const store = ResourceStore.create(
      {},
      {
        api,
        kindStore: KindStore.create({}),
        catalogStore: CatalogStore.create({}),
        categoryStore: CategoryStore.create({}, { api })
      }
    );
    const component = shallow(<App store={store} />);

    expect(component.find(Footer).length).toEqual(1);
  });
});
