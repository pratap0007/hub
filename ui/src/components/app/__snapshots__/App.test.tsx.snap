// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pf-c-page"
>
  <header
    className="pf-c-page__header"
  >
    <div
      className="pf-c-page__header-brand"
    >
      <a
        className="pf-c-page__header-brand-link"
        href="/"
        target=""
      >
        <img
          alt="Tekton Hub Logo"
          className="pf-c-brand"
          src="logo.png"
        />
      </a>
    </div>
    <div
      className="pf-c-page__header-tools"
    >
      <div
        className="pf-c-page__header-tools-group pf-m-hidden pf-m-visible-on-md"
      >
        <div
          className="pf-c-page__header-tools-item"
        >
          <span
            style={
              Object {
                "fontSize": "1em",
                "fontWeight": "bold",
                "marginRight": "1em",
              }
            }
          >
             
            Login
          </span>
        </div>
        <div
          className="pf-c-page__header-tools-item"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1.5em"
            role="img"
            style={
              Object {
                "marginRight": "1em",
              }
            }
            viewBox="0 0 448 512"
            width="1.5em"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            />
          </svg>
        </div>
        <div
          className="pf-c-page__header-tools-item"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1.5em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.1875em",
              }
            }
            viewBox="0 0 496 512"
            width="1.5em"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </div>
      </div>
    </div>
  </header>
  <main
    className="pf-c-page__main"
    tabIndex={-1}
  >
    <div
      className="background-image"
    >
      <h1
        className="heading"
      >
        Welcome to Tekton Hub
      </h1>
      <h3
        className="tag-line"
      >
        Discover, search and share reusable Tasks and Pipelines
      </h3>
    </div>
    <section
      className="pf-c-page__main-section"
      style={
        Object {
          "height": "80em",
        }
      }
    >
      <div
        className="pf-l-grid pf-m-gutter"
        rowSpan={12}
      >
        <div
          className="pf-l-grid__item pf-m-2-col pf-m-12-row"
        >
          Filter
        </div>
        <div
          className="pf-l-grid__item pf-m-10-col pf-m-12-row"
        >
          Resources
          <div
            className="pf-c-pagination pf-m-compact"
            data-ouia-component-id="OUIA-Generated-Pagination-top-1"
            data-ouia-component-type="PF4/Pagination"
            data-ouia-safe={true}
            id="pagination-options-menu-0"
          >
            <div
              className="pf-c-pagination__total-items"
            >
              <b>
                1
                 - 
                20
              </b>
               
              of 
              <b>
                200
              </b>
               
              
            </div>
            <div
              className="pf-c-options-menu"
              data-ouia-component-type="PF4/PaginationOptionsMenu"
              data-ouia-safe={true}
            >
              <div
                className="pf-c-options-menu__toggle pf-m-plain pf-m-text"
              >
                <span
                  className="pf-c-options-menu__toggle-text"
                >
                  <b>
                    1
                     - 
                    20
                  </b>
                   
                  of 
                  <b>
                    200
                  </b>
                   
                  
                </span>
                <button
                  aria-expanded={false}
                  aria-label="Items per page"
                  className="  pf-c-options-menu__toggle-button"
                  disabled={false}
                  id="pagination-options-menu-toggle-0"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <span
                    className="pf-c-options-menu__toggle-button-icon"
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <nav
              aria-label="Pagination"
              className="pf-c-pagination__nav"
            >
              <div
                className="pf-c-pagination__nav-control"
              >
                <button
                  aria-disabled={true}
                  aria-label="Go to previous page"
                  className="pf-c-button pf-m-plain pf-m-disabled"
                  data-action="previous"
                  data-ouia-component-id="OUIA-Generated-Button-plain-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe={true}
                  disabled={true}
                  onClick={[Function]}
                  tabIndex={null}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                    />
                  </svg>
                </button>
              </div>
              <div
                className="pf-c-pagination__nav-control"
              >
                <button
                  aria-disabled={false}
                  aria-label="Go to next page"
                  className="pf-c-button pf-m-plain"
                  data-action="next"
                  data-ouia-component-id="OUIA-Generated-Button-plain-2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe={true}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </button>
              </div>
            </nav>
          </div>
        </div>
      </div>
    </section>
    <section
      className="pf-c-page__main-section"
    >
      <article
        className="pf-c-card"
        data-ouia-component-id="OUIA-Generated-Card-1"
        data-ouia-component-type="PF4/Card"
        data-ouia-safe={true}
      >
        <div
          className="pf-l-grid"
        >
          <div
            className="pf-l-grid__item pf-m-4-col"
          />
          <div
            className="pf-l-grid__item pf-m-4-col pf-m-12-row"
          >
            <div
              className="pf-l-flex"
            >
              <div
                className=""
              >
                <a
                  href="https://cd.foundation"
                >
                  <img
                    alt="tekton.dev"
                    src="https://tekton.dev/partner-logos/cdf.png"
                  />
                </a>
              </div>
            </div>
            <div
              className="pf-l-flex"
              style={
                Object {
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="pf-c-content"
              >
                <h1
                  className=""
                  data-pf-content={true}
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Tekton is a
                   
                  <a
                    className=""
                    data-pf-content={true}
                    href="https://cd.foundation"
                  >
                    Continuous Delivery Foundation
                  </a>
                   
                  project.
                </h1>
              </div>
            </div>
            <div
              className="pf-l-flex"
              style={
                Object {
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="logo-size"
              >
                <img
                  alt="Tekton"
                  src="tekton.png"
                />
              </div>
            </div>
            <div
              className="pf-l-flex"
            >
              <div
                className="pf-c-card__footer"
              >
                <p
                  className=""
                  data-pf-content={true}
                  style={
                    Object {
                      "color": "white",
                      "textAlign": "center",
                    }
                  }
                >
                  © 2020 The Linux Foundation®. All rights reserved. The Linux Foundation has registered trademarks and uses trademarks. For a list of trademarks of The Linux Foundation, please see our
                   
                  <a
                    className=""
                    data-pf-content={true}
                    href="https://www.linuxfoundation.org/trademark-usage/"
                  >
                    Trademark Usage page
                  </a>
                  . Linux is a registered trademark of Linus Torvalds.
                   
                  <a
                    className=""
                    data-pf-content={true}
                    href="https://www.linuxfoundation.org/privacy/"
                  >
                    Privacy Policy
                  </a>
                   
                  and
                   
                  <a
                    className=""
                    data-pf-content={true}
                    href="https://www.linuxfoundation.org/terms/"
                  >
                    Terms of Use
                  </a>
                   
                  .
                </p>
              </div>
            </div>
          </div>
          <div
            className="pf-l-grid__item pf-m-4-col"
          />
        </div>
      </article>
    </section>
  </main>
</div>
`;
